src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -1173,7 +1119,14 @@ public function isNotFound()
      */
     public function isRedirect($location = null)
     {
-        return in_array($this->statusCode, array(201, 301, 302, 303, 307, 308)) && (null === $location ?: $location == $this->headers->get('Location'));
+        return in_array($this->statusCode, array(
+            self::HTTP_CREATED,
+            self::HTTP_MOVED_PERMANENTLY,
+            self::HTTP_FOUND,
+            self::HTTP_SEE_OTHER,
+            self::HTTP_TEMPORARY_REDIRECT,
+            self::HTTP_PERMANENTLY_REDIRECT
```

Comment:
The point of course is: why do we care about the numbers themselves? What we care about is parity. If the arrays and code use the constants, and there is a problem, then it is in a constant mapping, not a discrepancy between the scalar usage and the constant usage.